modules/rules_python/0.4.0/presubmit.yml

```diff
@@ -0,0 +1,17 @@
+build_targets: &build_targets
+- '@rules_python//python/pip_install/extract_wheels'
```

Comment:
rules_python already integration tests the release artifact it pushes to github, using bazel_integration_test within the repo.

would we want to repeat that testing here in the registry? I would expect that registry testing for releases is a poor substitute for the rules testing that themselves, since you don't notice a problem until you send a PR to this repo (which is probably after the release was already published)

Reply:
I had some explanation for this file here: https://github.com/bazelbuild/bazel-central-registry/tree/maintainer#presubmityml

> rules_python already integration tests the release artifact it pushes to github, using bazel_integration_test within the repo.

That is good to known as I just noticed you also strip out some sources in the release archive.

> would we want to repeat that testing here in the registry?

In rules_python's case, I think it's safe to exclude most of your test cases (they are not available in the release archive anyway). But it's still good to specify some build targets that you want to expose to users here, just as a sanity check and make sure nothing serious was going wrong. 

> I would expect that registry testing for releases is a poor substitute for the rules testing that themselves

Again, we don't intend this to be a substitution for your own testing.

The presubmit.yml probably makes more sense for you in the following case:

Imagine there is a project X, which depends on rules_python. X has a presubmit.yml to specify some heavily used build & test targets. When rules_python has a new release, we can upgrade rules_python version for X to see if those targets still pass. If not, either we discovered a bug in rules_python or we can notify the downstream user to migrate for the incompatible change. This can even be done before you release the new version by creating a new branch of this registry.

Reply:
okay, thanks for explaining. My only remaining concern is who is going to fix that patch file when you get inevitable conflicts from upstream. Or do you plan to send a PR to rules_python?

Reply:
Oh, yes, this will be upstreamed to rules_python definitely. It's just I'm patching 0.4.0 so I'm storing it in the registry for now.

My plan is to write some migration guide/tool to Bzlmod first, then the community can help migrating together, of course we'll be helping as well by sending PRs.